sdk/core/core/src/az_span.c

```diff
@@ -234,7 +234,7 @@ az_span az_span_copy(az_span destination, az_span source)
   // Implementations of memmove generally do the right thing when number of bytes to move is 0, even
   // if the ptr is null, but given the behavior is documented to be undefined, we disallow it as a
   // precondition.
-  AZ_PRECONDITION_VALID_SPAN(source, 0, false);
+  AZ_PRECONDITION_VALID_SPAN(source, 0, true);
```

Comment:
Please revert this change. This check should be made by the caller and they should not pass in `NULL` ptr to `az_span_copy`. See the comment right above this line.

The fix doesn't belong here but rather in `az_http_request_append_header`.

Reply:
well, this was @JeffreyRichter request actually
Since we should let customers call az_span_copy with NULL as sorce and that would just return dst without doing anything else.

Reply:
I agree that copying an **empty** span as the source into a destination should just return dst without doing anything else (for example a slice of a valid buffer that happens to have size 0). I am not sure that **a span with a null ptr** should be allowed to be passed in to `az_span_copy` though (such as `AZ_SPAN_NULL`).

And if we really want to support that here, in az_span_copy, we need to also add a check in `az_span_copy` to handle null ptr OR update the comment that warns against passing in null to `memmove`:

```C
uint8_t* ptr = az_span_ptr(source);
if (ptr == NULL)
{
   return destination;
}
```

Reply:
I will add that if condition.
We need to support it because this can happen on runtime (where preconditions have might been compiled away)
Imagine this
```c
az_span some_span = {0};
// Here customer call some function from their side to read/get some az_span from I/O
some_span = customer_funct(io);  // Even if this is consider to never fail, we can't be sure `some_span` will be updated to point somewhere..

// then customer try to copy 
az_span_copy(other_buffer, some_span);  // we can't tell customer to check that `some_span` is not null, because it can be runtime-calculated
```
 
```

Reply:
Couldn't the same be said about the destination? Why is destination not allowed to be `AZ_SPAN_NULL`?